sdk/servicebus/Azure.Messaging.ServiceBus/src/Processor/ProcessSessionMessageEventArgs.cs

```diff
@@ -30,7 +30,15 @@ public class ProcessSessionMessageEventArgs : EventArgs
         /// </summary>
         public CancellationToken CancellationToken { get; }
 
-        internal ConcurrentDictionary<ServiceBusReceivedMessage, byte> Messages => _receiveActions.Messages;
+        /// <summary>
+        /// The <see cref="System.Threading.CancellationToken"/> instance is cancelled when the lock renewal failed to
+        /// renew the lock or the <see cref="ServiceBusSessionProcessorOptions.MaxAutoLockRenewalDuration"/> has elapsed.
+        /// </summary>
+        /// <remarks>The cancellation token is triggered by comparing <see cref="ServiceBusReceivedMessage.LockedUntil"/>
+        /// against <see cref="DateTimeOffset.UtcNow"/> and might be subjected to clock drift.</remarks>
+        public CancellationToken LockExpiryCancellationToken { get; }
```

Comment:
Do we need the same for ProcessSessionEventArgs?

Reply:
Yeah I think that would make sense since the lock is at the session level.